packages/vite-plugin-checker/src/checkers/vueTsc/languagePlugins.js

```diff
@@ -0,0 +1,36 @@
+const vue = require('@vue/language-core')
+const { removeEmitGlobalTypes } = require('vue-tsc')
+
+const windowsPathReg = /\\/g
+
+// #region copied from https://github.com/vuejs/language-tools/blob/0781998a29f176ad52c30d3139d5c78a5688bd5d/packages/tsc/index.ts
+exports.getLanguagePlugins = (ts, options) => {
+  const { configFilePath } = options.options
+  const vueOptions =
+    typeof configFilePath === 'string'
+      ? vue.createParsedCommandLine(ts, ts.sys, configFilePath.replace(windowsPathReg, '/'))
+          .vueOptions
+      : vue.resolveVueCompilerOptions({})
+  const writeFile = options.host.writeFile.bind(options.host)
+  options.host.writeFile = (fileName, contents, ...args) => {
+    return writeFile(fileName, removeEmitGlobalTypes(contents), ...args)
+  }
+  const vueLanguagePlugin = vue.createVueLanguagePlugin(
+    ts,
+    (id) => id,
+    () => '',
+    (fileName) => {
+      const fileMap = new vue.FileMap(options.host?.useCaseSensitiveFileNames?.() ?? false)
+      for (const vueFileName of options.rootNames.map((rootName) =>
+        rootName.replace(windowsPathReg, '/')
+      )) {
+        fileMap.set(vueFileName, undefined)
+      }
+      return fileMap.has(fileName)
```

Comment:
Why not return directly in for?
I think it's will be faster
```
return fileMap.has(fileName) || options.rootNames.some(rootName => rootName.replace(windowsPathReg, '/') === fileName)
```

Reply:
Yeah but I'd like to keep it synched with upstream

Reply:
Sure, will do later

Reply:
I no longer use ts for that file(again) because tsup runs builds in parallel and I can't control who runs first, causing build failures. Please let me know if you've figured out a solution.